tests/addresslib/address_test.py

```diff
@@ -147,52 +147,81 @@ def test_display_name__to_full_spec():
         EmailAddress(u'Привет Медвед', 'foo@bar.com').full_spec())
 
 
-def test_address_str():
-    eq_(str(EmailAddress('foo <foo@bar.com>'        )), 'foo@bar.com')
-    eq_(str(EmailAddress('аджай <аджай@экзампл.рус>')), 'аджай@экзампл.рус')
-
-
-def test_address_repr():
-    eq_(repr(EmailAddress('foo <foo@bar.com>'        )), 'foo <foo@bar.com>')
-    eq_(repr(EmailAddress('аджай <аджай@экзампл.рус>')), 'аджай <аджай@экзампл.рус>')
-
-
-def test_address_full_spec():
-    eq_(EmailAddress('foo <foo@bar.com>').full_spec(), 'foo <foo@bar.com>')
-
-
-def test_address_full_spec_non_ascii_display_name():
-    eq_(EmailAddress('аджай <foo@bar.com>').full_spec(), '=?utf-8?b?0LDQtNC20LDQuQ==?= <foo@bar.com>')
-
-
-def test_address_full_spec_non_ascii_domain():
-    eq_(EmailAddress('foo <foo@экзампл.рус>').full_spec(), 'foo <foo@xn--80aniges7g.xn--p1acf>')
-
-
-def test_address_full_spec_non_ascii_local_part():
-    with assert_raises(ValueError):
-        EmailAddress('foo <аджай@bar.com>').full_spec()
-
-
-def test_address_to_unicode():
-    eq_(EmailAddress('foo <foo@bar.com>'        ).to_unicode(), u'foo <foo@bar.com>')
-    eq_(EmailAddress('аджай <аджай@экзампл.рус>').to_unicode(), u'аджай <аджай@экзампл.рус>')
+def test_views():
+    for i, tc in enumerate([{
+        # Pure ASCII
+        'addr': parse('foo <foo@bar.com>'),
+        'repr': 'foo <foo@bar.com>',
+        'str': 'foo@bar.com',
+        'unicode': u'foo <foo@bar.com>',
+        'full_spec': 'foo <foo@bar.com>',
```

Comment:
Understood. I just don't want to be the one that breaks it later by updating one and forgetting about the other.